specification/cognitiveservices/OpenAI.Inference/models/completions/streaming/chat_completions.tsp

```diff
@@ -0,0 +1,126 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@typespec/versioning";
+
+import "../azure_chat_extensions.tsp";
+import "../common.tsp";
+import "../functions.tsp";
+import "../chat_completions.tsp";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using TypeSpec.Versioning;
+
+namespace Azure.OpenAI;
+
+@doc("A single, role-attributed message within a chat completion interaction.")
+model ChatMessageStreamChunk {
+  @doc("The role associated with this message payload.")
+  @projectedName("json", "role")
+  role: ChatRole;
```

Comment:
I believe this can be `role?`, as it typically only appears in the first delta payload for a typical streamed choice.

E.g. (abbreviated from [a test assets file](https://github.com/Azure/azure-sdk-assets/blob/net/openai/Azure.AI.OpenAI_41a964dc8d/net/sdk/openai/Azure.AI.OpenAI/tests/SessionRecords/OpenAIInferenceTests/StreamingChatCompletions(Azure).json#L56)):

```
  "data: {...,"choices":[{"index":0,"finish_reason":null,"delta":{"role":"assistant"},...}\n\n",
  "data: {...,"choices":[{"index":0,"finish_reason":null,"delta":{"content":"A"}}\n\n",
```